runsc/container/container_test.go

```diff
@@ -3135,3 +3136,129 @@ func TestOverlayByMountAnnotation(t *testing.T) {
 		t.Fatalf("overlay filestore at %q was not deleted after container.Destroy()", filestoreFile)
 	}
 }
+
+// TestMountEROFS checks that the checksums from the target directory in container
+// are identical with the ones from the source directory on host.
+func TestMountEROFS(t *testing.T) {
+	// Create a temporary directory with some random files in it, which will
+	// be used as the source directory to create the EROFS images.
+	sourceDir, err := ioutil.TempDir(testutil.TmpDir(), "erofs-source")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir() failed: %v", err)
+	}
+	testApp, err := testutil.FindFile("test/cmd/test_app/test_app")
+	if err != nil {
+		t.Fatalf("error finding test_app: %v", err)
+	}
+
+	// Source directory is a small directory. Let's create a big directory in it.
+	cmd := fmt.Sprintf("%s fsTreeCreate --target-dir=%s --create-symlink --depth=1 --file-per-level=500 --file-size=5000", testApp, filepath.Join(sourceDir, "big-directory"))
+	if out, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput(); err != nil {
+		t.Fatalf("exec: sh -c %q, err: %v, out: %s", cmd, err, out)
+	}
+
+	// Create a test script which can be used to get the checksums
+	// from a specified directory.
+	scriptFile := path.Join(sourceDir, "test-script")
```

Comment:
scriptFile is created inside `erofs-source` itself? Maybe better to create a separate tmp file with this script?

Reply:
My idea is to use scriptFile as an input for the test as well. But it's not really necessary. Will create a separate tmp file.